airflow/api_connexion/endpoints/log_endpoint.py

```diff
@@ -15,12 +15,93 @@
 
+from flask import Response, current_app, request
+from itsdangerous.exc import BadSignature
+from itsdangerous.url_safe import URLSafeSerializer
 
-def get_log():
+from airflow.api_connexion.exceptions import BadRequest, NotFound
+from airflow.api_connexion.schemas.log_schema import logs_schema
+from airflow.models import DagRun
+from airflow.utils.log.log_reader import TaskLogReader
+from airflow.utils.session import provide_session
+
+
+@provide_session
+def get_log(session, dag_id, dag_run_id, task_id, task_try_number,
+            full_content=False, token=None):
     """
     Get logs for specific task instance
     """
-    raise NotImplementedError("Not implemented yet.")
+    if not token:
+        metadata = {}
+    else:
+        key = current_app.config["SECRET_KEY"]
+        try:
+            metadata = URLSafeSerializer(key).loads(token)
+        except BadSignature:
+            raise BadRequest("Bad Signature. Please sign your token with URLSafeSerializer")
+
+    if metadata.get('download_logs', None) and metadata['download_logs']:
+        full_content = True
+
+    if full_content:
+        metadata['download_logs'] = True
+    else:
+        metadata['download_logs'] = False
+
+    task_log_reader = TaskLogReader()
+    if not task_log_reader.is_supported:
+        metadata = {"end_of_log": True}
+        return logs_schema.dump(dict(
+            content="[Task log handler does not support read logs.]",
+            continuation_token=str(metadata)
+        ))
+
+    query = session.query(DagRun).filter(DagRun.dag_id == dag_id)
+    dag_run = query.filter(DagRun.run_id == dag_run_id).first()
+    if not dag_run:
+        raise NotFound("Specified DagRun not found")
+
+    ti = dag_run.get_task_instance(task_id, session)
+    if ti is None:
+        metadata['end_of_log'] = True
+        return logs_schema.dump(
+            dict(
+                content="[*** Task instance did not exist in the DB\n]",
+                continuation_token=str(metadata))
+        )
+    try:
+        dag = current_app.dag_bag.get_dag(dag_id)
+        if dag:
+            ti.task = dag.get_task(ti.task_id)
+
+        if request.headers.get('Content-Type', None) == 'application/json':
+            logs, metadata = task_log_reader.read_log_chunks(ti, task_try_number, metadata)
+            logs = logs[0] if task_try_number is not None else logs
+
+            return logs_schema.dump(dict(continuation_token=str(metadata),
+                                         content=logs)
+                                    )
+
+        # Defaulting to content type of text/plain
+        logs = task_log_reader.read_log_stream(ti, task_try_number, metadata)
+
+        attachment_filename = task_log_reader.render_log_filename(ti, task_try_number)
+
+        return Response(
+            logs,
+            mimetype="text/plain",
+            headers={
+                "Content-Disposition": f"attachment; filename={attachment_filename}"
```

Comment:
It should open in your browser to make it easier to view. Kubernetes API works in a similar way.

Reply:
I am not sure, how I can test that it opens in the browser but take a look at how I did it.

Reply:
you can check the interaction with the browser by doing manual tests. :-D